.github/workflows/unit-test-main.yml

```diff
@@ -0,0 +1,13 @@
+name: Run tests on `main` push
```

Comment:
suggestion (non-blocking): Also, since the badge has the form "<name> | passing", perhaps it would be better to name this workflow just "Tests on `main` push"?